Packages/Netherlands3D/SelectionTools/Runtime/Scripts/SelectionTools/AreaSelection.cs

```diff
@@ -0,0 +1,205 @@
+*  You may obtain a copy of the License at:
+*
+*    https://github.com/Amsterdam/3DAmsterdam/blob/master/LICENSE.txt
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" basis,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+*  implied. See the License for the specific language governing
+*  permissions and limitations under the License.
+*/
+using Netherlands3D.Events;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Netherlands3D.SelectionTools
+{
+    public class AreaSelection : MonoBehaviour
+    {
+        [Header("Input")]
+        [SerializeField] private InputActionAsset inputActionAsset;
+        private InputActionMap areaSelectionActionMap;
+        private MeshRenderer boundsMeshRenderer;
+
+        [Header("Invoke")]
+        [SerializeField] private BoolEvent selectingArea;
+        [SerializeField] private BoundsEvent selectedAreaBounds;
+
+        [Header("Settings")]
+        [SerializeField] private float gridSize = 100;
+        [SerializeField] private float multiplyHighlightScale = 5.0f;
+        [SerializeField] private float maxSelectionDistanceFromCamera = 10000;
+        [SerializeField] private bool useWorldSpace = false;
+
+        private InputAction pointerAction;
+        private InputAction tapAction;
+        private InputAction clickAction;
+        private InputAction modifierAction;
+
+        private Vector3 selectionStartPosition;
+
+        private Plane worldPlane;
+
+        [SerializeField] private GameObject gridHighlight;
+        [SerializeField] private GameObject selectionBlock;
+        [SerializeField] private Material triplanarGridMaterial;
+
+        private bool drawingArea = false;
+
+        void Awake()
+        {
+            areaSelectionActionMap = inputActionAsset.FindActionMap("AreaSelection");
+            tapAction = areaSelectionActionMap.FindAction("Tap");
+            clickAction = areaSelectionActionMap.FindAction("Click");
+            pointerAction = areaSelectionActionMap.FindAction("Point");
+            modifierAction = areaSelectionActionMap.FindAction("Modifier");
+
+            tapAction.performed += context => Tap();
+            clickAction.performed += context => StartClick();
+            clickAction.canceled += context => Release();
+
+            worldPlane = (useWorldSpace) ? new Plane(Vector3.up, Vector3.zero) : new Plane(this.transform.up, this.transform.position);
+
+            boundsMeshRenderer = selectionBlock.GetComponent<MeshRenderer>();
+        }
+
+        private void OnValidate()
+        {
+            selectionBlock.transform.localScale = Vector3.one * gridSize;
+            gridHighlight.transform.localScale = Vector3.one * gridSize * multiplyHighlightScale;
+            triplanarGridMaterial.SetFloat("GridSize", 1.0f / gridSize);
+        }
+
+        private void OnEnable()
```

Comment:
bij onEnable of OnDisable misschien het selectieblok resetten? als je nu voor een tweede keer een selectie maakt zie je je oude selectie nog.